build.js

```diff
@@ -1,21 +1,62 @@
-const fs = require('fs');
-const yaml = require('yaml');
+const assert = require("assert");
+const fs = require("fs");
+const yaml = require("yaml");
+
+let flowzoneSource = fs.readFileSync("./flowzone.yml", "utf8");
+
+// Pattern to find template variables (sub lines with file)
+const templateVariable = (p) => {
+  // Tab or spaced comment with curly braces enclosing a variable
+  return new RegExp(`[ \t]*# {{ *${p} *}}`, "g");
+};
+const fileTemplate = (p = "(.*[a-zA-Z0-9/_-]+.yml)") => {
+  return templateVariable(p);
+};
+const matches = flowzoneSource.matchAll(fileTemplate());
+if (matches) {
+  // Replace any template variables before passing to yaml exploder
+  for (const m of matches) {
+    const entireLine = m[0];
+    const spacing = entireLine.substring(0, entireLine.indexOf("#"));
+    const templateValue = fs.readFileSync(m[1], "utf8");
+    const sub =
+      spacing.length > 0 // Adding template tabs/spaces to template source for proper indenting
+        ? tabText(templateValue, spacing.length)
+        : templateValue;
+    flowzoneSource = flowzoneSource.replaceAll(fileTemplate(m[1]), sub);
+  }
+}
+
+function tabText(text, numOfTabs) {
```

Comment:
maybe confusing name since yaml does not allow tabs.

Reply:
yeah, the terminology you want is `indentText` where indentation can be either tabs or space in general (although yaml enforces spaces)